src/transformers/SingleLoopTransformation.h

```diff
@@ -70,9 +70,11 @@ class SingleLoopTransformation {
         VerificationResult translate(TransitionSystemVerificationResult result);
 
     private:
-        InvalidityWitness translateErrorPath(std::size_t unrolling);
+        template<typename T> using ErrorOr = std::variant<NoWitness, T>;
```

Comment:
Name is a little bit weird
Maybe `EmptyOrWitness` or smth?

Reply:
`ErrorOr<Witness>` captures the semantic I wanted to express. I just did not want to write the whole templated class for that.
The functions where it is used return `Validity/InvalidityWitness` under normal circumstances, but in some cases they may not be able to compute the witness, which is the `Error` result.